### PR TITLE
Add Blackbox monitoring for backend endpoints

### PR DESCRIPTION
**Pull request by Alex-Welsh**
Adding backend blackbox endpoint monitoring.

Some service do not support TLS at all (e.g. Blazar)
Some services do support TLS, and they have a globally-scoped variable (e.g. Neutron, `neutron_enable_tls_backend`)
Some services do support TLS, but have locally scoped variables, so must use the generic `kolla_enable_tls_backend` (e.g. Nova)

This is a precursor to proper support in Kolla-Ansible, which I'm planning on doing with file-based service discovery

### CODE REVIEW

**Comment by Alex-Welsh** [comment]
For context, this has been tested in a Multinode.

Confirmed working:

-  barbican
-  cinder
-  designate
-  glance
-  heat
-  heat_cfn
-  horizon
-  magnum
-  neutron
-  nova
-  octavia
-  placement

Untested (not deployed by multinode):
- blazar
- gnocchi
- manila
- ironic
- ironic inspector
- swift

Example templated content:
``` yaml
targets:
    - cinder_backend_rockyovnfqdncontroller01:os_endpoint:https://192.168.37.3:8776
    - cinder_backend_rockyovnfqdncontroller02:os_endpoint:https://192.168.37.4:8776
    - cinder_backend_rockyovnfqdncontroller03:os_endpoint:https://192.168.37.5:8776
    - designate_backend_rockyovnfqdncontroller01:os_endpoint:http://192.168.37.3:9001
    - designate_backend_rockyovnfqdncontroller02:os_endpoint:http://192.168.37.4:9001
    - designate_backend_rockyovnfqdncontroller03:os_endpoint:http://192.168.37.5:9001

```

**Comment by Alex-Welsh** [comment]
Updated to include a warning about having separate monitoring nodes

File: releasenotes/notes/blackbox-backend-4415919e10a1aa4e.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    Added Blackbox monitoring for backend endpoints by default. Note that this
+    configuration will only work if the Blackbox exporters have access to the
+    backend endpoints.
+issues:
+  - |
+    Backend Blackbox monitoring will not work if the exporter does not have
+    access to the backend OpenStack endpoints. This ususally happens when
+    separate monitoring nodes are deployed. In this case, move the Blackbox
+    exporter to the Haproxy group, remove the endpoints from
+    `etc/kayobe/kolla/inventory/group_vars/prometheus-Blackbox-exporter`, or
```
Comment (MoteHue) [inline releasenotes/notes/blackbox-backend-4415919e10a1aa4e.yaml]:
```suggestion
    `etc/kayobe/kolla/inventory/group_vars/prometheus-blackbox-exporter`, or
```

Comment (Alex-Welsh) [reply releasenotes/notes/blackbox-backend-4415919e10a1aa4e.yaml]: too casual with find and replace it seems

Comment (Alex-Welsh) [reply releasenotes/notes/blackbox-backend-4415919e10a1aa4e.yaml]: Fixed

**Review by dougszumski — DISMISSED** [review]
Thanks Alex!

File: releasenotes/notes/blackbox-backend-4415919e10a1aa4e.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+features:
+  - |
+    Added Blackbox monitoring for backend endpoints by default. Note that this
+    configuration will only work if the Blackbox exporters have access to the
+    backend endpoints.
+issues:
+  - |
+    Backend Blackbox monitoring will not work if the exporter does not have
+    access to the backend OpenStack endpoints. This ususally happens when
```
Comment (dougszumski) [inline releasenotes/notes/blackbox-backend-4415919e10a1aa4e.yaml]: micronit: ususally

Comment (Alex-Welsh) [reply releasenotes/notes/blackbox-backend-4415919e10a1aa4e.yaml]: I can't leave a spelling mistake in good conscience :laughing:

**Comment by Alex-Welsh** [comment]
@dougszumski I've fixed that typo, could you do a quick re-review?

**Review by dougszumski — APPROVED** [review]
Thanks Alex!

**Pull request merged**